src/main.rs

```diff
@@ -0,0 +1,19 @@
+use std::path::Path;
+
+use ssufid::{core::SsufidCore, plugins::example::ExamplePlugin};
+use tokio::io::AsyncWriteExt;
+
+#[tokio::main]
+async fn main() -> eyre::Result<()> {
+    let mut core = SsufidCore::new("./.ssufid/cache");
+
+    let site = core.run(ExamplePlugin).await?;
+
+    let out_dir = Path::new("./out/example");
+    tokio::fs::create_dir_all(out_dir).await?;
+
+    let mut example_file = tokio::fs::File::create_new(out_dir.join("data.json")).await?;
+    let example_json = serde_json::to_string_pretty(&site)?;
+    example_file.write_all(example_json.as_bytes()).await?;
```

Comment:
p3) 합의된 아키텍처에 대해 제가 기억하기론, daemon에서는 core run 호출하고 나온 결과(`SsufidSiteData` 타입)를 RSS로 변환하고 저장만 하는 걸로 알고 있어요. 한번 확인해주시면 감사하겠습니다. (작성해주신 로직은 캐시 데이터를 저장하는 것 같아요)

예시:
```suggestion
    let mut example_file = tokio::fs::File::create_new(out_dir.join("data.json")).await?;
    let rss_str = site.to_rss();
    example_file.write_all(rss_str.as_bytes()).await?;
```

Reply:
제 기억으로는 json 데이터를 그대로 저장하고, rss도 같이 저장하는 걸로 이해했는데요(rss 형식 없이 raw data도 파싱할 수 있도록?).

`to_rss()`는 제가 구현에 넣지 않은게 맞습니다(todo때문에 패닉). 주석으로 추가해둘게요.

별개로 json 데이터도 같이 접근할 수 있도록 하는게 유용하지 않을까요?

Reply:
> 코드 상의 리뷰에도 달아뒀듯이 rss 말고 json 형태로도 데이터를 제공해야 하지 않을까라는 이야기에요. 생각해보시고 답 달아주심 될듯

아하 맞네요 이해했습니다!

p3) json으로 변환하는 것도 한 줄이긴 하지만 `to_rss()`랑 패턴을 맞추기 위해 `SsufidSiteData`에 `to_json()` 메서드를 추가하는 건 어떨까요?

Reply:
> to_rss()랑 패턴을 맞추기 위해 SsufidSiteData에 to_json() 메서드를 추가하는 건 어떨까요?

저는 `to_rss()` 랑 `to_json()`은 레이어가 다르다고 생각해요. SiteData -> RSS는 data structure 간의 변환이고(Ssufid 데이터를 RSS spec에 맞게 변환), json은 data representation이니까요.

그래서 to_rss도 RSS 구조를 표현하는 struct(external crate에서든 자체 구현이든)로 변환하고 serde_xml 등으로 serialization 하여 저장하는게 이상적이다 생각해요(다만 과도기적 구현으로 String으로 output하는것도 괜찮다고 생각)

Reply:
아아 넵 그게 확실히 더 깔끔하겠네요. 좋습니다👍